src/frontend/planner_test/tests/testdata/input/subquery.yaml

```diff
@@ -226,3 +226,31 @@
     select *, (select sum(i) from (select unnest(x) i) Q where k = i ) as sum_x from t;
   expected_outputs:
     - optimizer_error
+- name: test over window subquery 1
+  sql: |
+    CREATE TABLE integers(i INTEGER);
+    SELECT i, (SELECT row_number() OVER (ORDER BY i) FROM integers WHERE i1.i=i limit 1) col FROM integers i1 ORDER BY i;
+  expected_outputs:
+    - batch_plan
+    - stream_plan
+- name: test over window subquery 2 (with nested loop join so cannot be transformed into a stream plan)
+  sql: |
+    CREATE TABLE integers(i INTEGER);
+    SELECT i1.i, (SELECT row_number() OVER (ORDER BY i) FROM integers WHERE i1.i=i limit 1) col FROM integers i1, integers i2 ORDER BY i1.i;
+  expected_outputs:
+    - batch_plan
+    - stream_error
+- name: test over window subquery 3
+  sql: |
+    CREATE TABLE integers(i INTEGER);
+    SELECT i, (SELECT SUM(i) OVER (ORDER BY i) FROM integers WHERE i1.i=i limit 1) col FROM integers i1 ORDER BY i;
```

Comment:
Why do all unit tests have a `limit 1`? Is it necessary?

Reply:
`Limit 1` is necessary in our system, because we check the scalar subquery which requires max-one-row property at compile time.